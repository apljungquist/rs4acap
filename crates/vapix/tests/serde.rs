use insta::assert_snapshot;
use rs4a_vapix::{
    action1::AddActionConfigurationResponse,
    apis,
    basic_device_info_1::AllUnrestrictedPropertiesData,
    json_rpc::parse_data,
    soap::parse_soap,
    soap_http::SoapRequest,
    system_ready_1::{EnglishBoolean, SystemreadyData},
};

#[test]
fn can_deserialize_action_1_examples() {
    let text =
        include_str!("../src/services/action1/examples/add_action_configuration_response.xml");
    let data = parse_soap::<AddActionConfigurationResponse>(text).unwrap();
    assert_eq!(data.configuration_id, 1);
}

#[test]
fn can_deserialize_basic_device_info_1_examples() {
    let text = include_str!(
        "../src/axis_cgi/basic_device_info_1/get_all_unrestricted_properties_200.json"
    );
    let data = parse_data::<AllUnrestrictedPropertiesData>(text).unwrap();
    assert_eq!(data.property_list.version, "12.5.56");
}

#[test]
fn can_deserialize_system_ready_1_examples() {
    let text = include_str!("../src/axis_cgi/system_ready_1/system_ready_200.json");
    let data = parse_data::<SystemreadyData>(text).unwrap();
    assert!(matches!(data.needsetup, EnglishBoolean::No));
}

#[test]
fn can_serialize_action_1_requests() {
    assert_snapshot!(
        apis::action_1::add_action_configuration("com.axis.action.fixed.ledcontrol")
            .name("Flash status LED")
            .param("led", "statusled")
            .param("color", "green,none")
            .param("duration", "1")
            .param("interval", "250")
            .to_envelope()
            .unwrap()
    );
    assert_snapshot!(apis::action_1::add_action_rule().to_envelope().unwrap());
    assert_snapshot!(apis::action_1::get_action_configurations()
        .to_envelope()
        .unwrap());
    assert_snapshot!(apis::action_1::get_action_rules().to_envelope().unwrap());
}
