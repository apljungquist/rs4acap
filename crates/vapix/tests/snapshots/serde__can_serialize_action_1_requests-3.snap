---
source: crates/vapix/tests/serde.rs
expression: "apis::action_1::get_action_configurations().to_envelope().unwrap()"
---
<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope"><soap:Body xmlns:tns1="http://www.onvif.org/ver10/topics" xmlns:tnsaxis="http://www.axis.com/2009/event/topics"><GetActionConfigurations xmlns="http://www.axis.com/vapix/ws/action1"/></soap:Body></soap:Envelope>
